repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -3087,12 +3091,23 @@ public void updateClassifications(EntityMutationContext context, String guid, Li
 
                 updatedTagPropagation = null;
             }
+            Boolean currentRestrictThroughLineage = currentClassification.getRestrictPropagationThroughLineage();
```

Comment:
Move this logic before task creation and create a new variable removeTagProp... and  pass it to the task as well as use it in this method to decide if we need to removeTagProp
No need to pass currentClassification.getRestrictPropagationThroughLineage() to task then

Reply:
Yeah, fixing it